### DISCUSSION TITLE
Feature Idee/Anfrage: Steuerung der max Ausgangsleistung von Zendure

### DISCUSSION THREAD

**Discussion by theMPstig** [hoylabs/OpenDTU-OnBattery#1306] [General]
Hallo zusammen,

erstmal danke für das coole Projekt hier. In meinem Setup nutze ich nicht den Powerlimiter von OpenDTUonBattery, da ich 2 PV-Module direkt am Wechselrichter habe und 4 am Zendure-Batteriesystem. Wenn ich den Powerlimiter nutzen würde, regelt er logischerweise ab, wenn mehr Strom produziert als verbraucht wird.
Ich möchte aber zumindest den Überschuss der beiden direkt am WR angeschlossenen Module immer sozusagen im Bypass direkt einspeisen. Hierfür würde es m.E. gehen, wenn man statt im Wechselrichter die Ausgangsleistung im Zendure-Hub den entsprechenden Wert setzt.
Angebunden ist das über MQTT (siehe hier: https://github.com/nograx/ioBroker.zendure-solarflow) und es gibt den Control-Wert "setOutputLimit". 
Besteht da irgendeine Möglichkeit zur Implementation hier oder ist zu sehr special Interest?
Danke und Grüße

**Comment by spcqike** [comment]
wenn ich dich richtig verstanden habe, ist es ein sehr spezieller use-case.

wenn die MQTT anbindung des zendure bereits über ioBroker läuft, kannst du nicht dort dein Limit berechnen und verschicken?

Hoymiles voll auf, ohne DPL, und dem Zendure als Limit dein ``istLimit+powerMeterReading`` übermitteln. steigt der Wert am powerMeter, wird das Limit erhöht. fällt er, wird es reduziert. hast du ein negativen powerMeterReading und kein aktives Limit, hast du Überschuss rein aus den zwei PV Modulen.

**Reply by vaterlangen** [reply]
Ein ähnliches Setup - nur ohne Vendor Cloud - habe ich auch laufen und eine Grundlegende Integration in OpenDTU-OnBattery entwickelt (siehe #1221 ). Aktuell kann man damit nur die Werte auslesen und ein paar Dinge einstellen. Ich suche aber bereits nach einer Möglichkeit, die Ausgangsleistung intelligent anzusteuern um bei Ausreichender Solarleistung direkt am WR keinen zusätzliche Leistung aus Solarflow zu ziehen. Ist aber - wie es Grade aussieht - nicht ganz trivial.

**Reply by theMPstig** [reply]
> wenn ich dich richtig verstanden habe, ist es ein sehr spezieller use-case.
> 
> wenn die MQTT anbindung des zendure bereits über ioBroker läuft, kannst du nicht dort dein Limit berechnen und verschicken?
> 
> Hoymiles voll auf, ohne DPL, und dem Zendure als Limit dein `istLimit+powerMeterReading` übermitteln. steigt der Wert am powerMeter, wird das Limit erhöht. fällt er, wird es reduziert. hast du ein negativen powerMeterReading und kein aktives Limit, hast du Überschuss rein aus den zwei PV Modulen.

Klingt logisch und gut. Habe ich noch gar nicht daran gedacht. Da ich in ioBroker aber totaler Anfänger bin, müsste ich mal schauen, ob ich das mit Blockly gelöst bekomme. Grds. wäre es mir in einer einfachen Konfiguration wie hier im Tool viel lieber. Aber ich dachte mir schon, dass es dafür wohl nicht zu viele Nutzer geben wird, für die das interessant ist.

**Reply by theMPstig** [reply]
> Ein ähnliches Setup - nur ohne Vendor Cloud - habe ich auch laufen und eine Grundlegende Integration in OpenDTU-OnBattery entwickelt (siehe #1221 ). Aktuell kann man damit nur die Werte auslesen und ein paar Dinge einstellen. Ich suche aber bereits nach einer Möglichkeit, die Ausgangsleistung intelligent anzusteuern um bei Ausreichender Solarleistung direkt am WR keinen zusätzliche Leistung aus Solarflow zu ziehen. Ist aber - wie es Grade aussieht - nicht ganz trivial.

Hui, das geht ja eigentlich genau in die Richtung. Die Zendure iobroker-Integration (Link oben) kann ja schon Werte steuern. Warum ist das dann hier komplizierter?

**Reply by vaterlangen** [reply]
Das ansteuern der Solarflow ist nicht das Problem - da läuft bei mir testweise schon eine Zeitgesteuerte Anpassung der Ausgangsleistung. Das Problem ist eher eine saubere Integration in DPL (erst direkte Panels, dann Anfrage an Solarflow nach mehr Leistung und umgekehrt)

**Reply by theMPstig** [reply]
Was ist jetzt nochmal DPL? Und wie machst du das zeitgesteuert? Hab mir grade mal ein script gebastelt. Ohne Zeitsteuerung und da wird dann natürlich sekündlich angepasst, was ja wenig Sinn macht...Ich dachte grad drüber nach, z.B. nur zu ändern, wenn der Wert um 50 Watt abweicht, aber vielleicht hast du eine schlauere Lösung?

**Reply by vaterlangen** [reply]
> Was ist jetzt nochmal DPL? 

Dynamic Power Limiter

> Und wie machst du das zeitgesteuert?

Aktuell noch ganz stupide nach Tag/Nacht. 
`t > Sonnenaufgang + x && t < Sonnenuntergang +y => Setzte Taglimit`
`t > Sonnenuntergang+y => Setzte Nachtlimit`

> vielleicht hast du eine schlauere Lösung?

Hysterese solltest du auf jeden Fall einbauen. Und bedenken, das Solarflow ziemlich langsam ist bei der Anpassung der Ausgangsleistung. Oder du nimmst https://github.com/reinhard-brandstaedter/solarflow-control und steuerst dann das WR Limit der OpenDTU darüber an.
